fdbserver/workloads/SidebandSingle.actor.cpp

```diff
@@ -0,0 +1,245 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbclient/Knobs.h"
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbserver/TesterInterface.actor.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+struct SidebandMessage {
+	constexpr static FileIdentifier file_identifier = 11862047;
+	uint64_t key;
+	Version commitVersion;
+
+	SidebandMessage() {}
+	SidebandMessage(uint64_t key, Version commitVersion) : key(key), commitVersion(commitVersion) {}
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, key, commitVersion);
+	}
+};
+
+struct SidebandInterface {
+	constexpr static FileIdentifier file_identifier = 15950545;
+	RequestStream<SidebandMessage> updates;
+
+	UID id() const { return updates.getEndpoint().token; }
+
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, updates);
+	}
+};
+
+struct SidebandSingleWorkload : TestWorkload {
+	double testDuration, operationsPerSecond;
+	SidebandInterface interf;
+
+	std::vector<Future<Void>> clients;
+	PerfIntCounter messages, consistencyErrors, keysUnexpectedlyPresent;
+
+	SidebandSingleWorkload(WorkloadContext const& wcx)
+	  : TestWorkload(wcx), messages("Messages"), consistencyErrors("Causal Consistency Errors"),
+	    keysUnexpectedlyPresent("KeysUnexpectedlyPresent") {
+		testDuration = getOption(options, LiteralStringRef("testDuration"), 10.0);
+		operationsPerSecond = getOption(options, LiteralStringRef("operationsPerSecond"), 50.0);
+	}
+
+	std::string description() const override { return "SidebandSingleWorkload"; }
+	Future<Void> setup(Database const& cx) override {
+		if (clientId != 0)
+			return Void();
+		return persistInterface(this, cx);
+	}
+	Future<Void> start(Database const& cx) override {
+		if (clientId != 0)
+			return Void();
+
+		clients.push_back(mutator(this, cx));
+		clients.push_back(checker(this, cx));
+		return delay(testDuration);
+	}
+	Future<bool> check(Database const& cx) override {
+		if (clientId != 0)
+			return true;
+		int errors = 0;
+		for (int c = 0; c < clients.size(); c++) {
+			errors += clients[c].isError();
+		}
+		if (errors)
+			TraceEvent(SevError, "TestFailure").detail("Reason", "There were client errors.");
+		clients.clear();
+		if (consistencyErrors.getValue())
+			TraceEvent(SevError, "TestFailure").detail("Reason", "There were causal consistency errors.");
+		return !errors && !consistencyErrors.getValue();
+	}
+
+	void getMetrics(std::vector<PerfMetric>& m) override {
+		m.push_back(messages.getMetric());
+		m.push_back(consistencyErrors.getMetric());
+		m.push_back(keysUnexpectedlyPresent.getMetric());
+	}
+
+	ACTOR Future<Void> persistInterface(SidebandSingleWorkload* self, Database cx) {
+		state Transaction tr(cx);
+		BinaryWriter wr(IncludeVersion());
```

Comment:
Can this use `ObjectWriter`?

Reply:
Looking at this closer, it appears this interface is just used on a single client. Could we avoid storing it in the database at all and instead model this as a `PromiseStream` that is stored in memory here?

Reply:
This is mostly code that was taken from the original `Sideband.actor.cpp` so I didn't bother to change too much other than restricting it to a single client. Is the concern here that it's more inefficient in the single-client scenario?

Reply:
No, just that the code would be a simpler if we weren't storing the interface in the cluster.

Reply:
I think this is worth hammering out, because if there's no need to do the interface shuffling it seems like we shouldn't add it. Perhaps there is some explanation for it that I'm unfamiliar with, though, do you know why we do this @sfc-gh-etschannen?

Also, if we do keep it we should try switching to `ObjectWriter`. `BinaryWriter` should only be used in special circumstances.

Reply:
Oh actually in the Sideband test, the clients are exchanging interfaces with each other, which is why they go in the database. Since you aren't doing that in this test, you shouldn't need to do that.

Reply:
Got it. In that case, I think that means we could just completely remove both `persistInterface` and `fetchSideband` and instead just keep the interface as a member in the workload class